src/storage/src/types/sources.rs

```diff
@@ -1732,6 +1734,50 @@ impl Codec for SourceData {
     }
 }
 
+/// Resource allocations for a storage instance.
+///
+/// Has some overlap with mz_controller::ComputeInstanceReplicaAllocation,
+/// but keeping it separate for now due slightly different semantics.
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct StorageInstanceResourceAllocation {
+    /// The memory limit for each process in the replica.
+    pub memory_limit: Option<MemoryLimit>,
+    /// The CPU limit for each process in the replica.
+    pub cpu_limit: Option<CpuLimit>,
+    /// The number of worker threads in the replica.
+    pub workers: NonZeroUsize,
+}
+
+impl StorageInstanceResourceAllocation {
+    pub fn new() -> Self {
```

Comment:
Can we give this a name that will discourage someone else from calling it when they ought to be doing something else?

```suggestion
    pub fn new_for_sinks_temp() -> Self {
```